libraries/ms-drivers/test/test_max6643_fan_controller.c

```diff
@@ -0,0 +1,99 @@
+#include "interrupt.h"
+#include "log.h"
+#include "max6643_fan_controller.h"
+#include "test_helpers.h"
+#include "unity.h"
+#include "wait.h"
+
+typedef struct Max6643MockStatus {
+  uint8_t FANFAIL_INTERRUPT_STATUS;  // 0 if not triggered, 1 otherwise
+  uint8_t OVERTEMP_INTERRUPT_STATUS;
+} Max6643MockStatus;
+
+static Max6643MockStatus s_mock_statuses;
+static Max6643Storage s_mock_storage;
+static Max6643Storage s_storage;
+
+static void prv_test_max6643_fanfail_interrupt_callback(const GpioAddress *address, void *context) {
+  LOG_DEBUG("FANFAIL INTERRUPT CALLBACK TRIGGERED\n");
+
+  TEST_ASSERT_EQUAL(1, s_mock_statuses.FANFAIL_INTERRUPT_STATUS);
+}
+
+static void prv_test_max6643_overtemp_interrupt_callback(const GpioAddress *address,
+                                                         void *context) {
+  LOG_DEBUG("OVERTEMP INTERRUPT CALLBACK TRIGGERED\n");
+
+  TEST_ASSERT_EQUAL(1, s_mock_statuses.OVERTEMP_INTERRUPT_STATUS);
+}
+
+void setup_test(void) {
+  gpio_init();
+  interrupt_init();
+  gpio_it_init();
+  s_mock_statuses.FANFAIL_INTERRUPT_STATUS = 0;
+  s_mock_statuses.OVERTEMP_INTERRUPT_STATUS = 0;
+}
+
+void teardown_test(void) {}
+
+// Test initializing with valid parameters
+void test_max6643_init_works(void) {
+  GpioAddress test_fanfail_pin = { .port = GPIO_PORT_A, .pin = 0 };
+  GpioAddress test_overtemp_pin = { .port = GPIO_PORT_A, .pin = 1 };
```

Comment:
To avoid redeclaring this at the start of each test you can make these static const variables.